server/cmwell-data-tools/src/main/scala/cmwell/tools/data/utils/akka/Retry.scala

```diff
@@ -186,18 +188,28 @@ object Retry extends DataToolsLogging with DataToolsConfig{
           e.discardBytes()
           Future.successful(Failure(new Exception(s"status is not success ($s) $e")) -> state.copy(response = response.toOption))
 
-        case (response@Success(res@HttpResponse(s, _, e, _)), state) =>
+        case (response@Success(res@HttpResponse(s, headers, e, _)), state) =>
           // consume HTTP response bytes and later pack them in fake response
-          val responseAndState = e.withoutSizeLimit().dataBytes.runFold(blank)(_ ++ _)
-            .map { dataBytes =>
-              val isValidBody = responseBodyValidator(dataBytes)
-
-              if (isValidBody) {
-                Success(res.copy(entity = dataBytes)) -> state.copy(response = response.toOption)
-              } else {
-                Failure(new Exception("response body is not valid")) -> state.copy(response = response.toOption)
+          val responseAndState = (e match {
+            case chunkedEntity:Chunked => {
+              chunkedEntity.chunks.runFold[ (ByteString,Option[Seq[HttpHeader]]) ]((ByteString.empty,None)) {
+                case (accumulatedEntity, chunk: LastChunk) => (accumulatedEntity._1 ++ chunk.data, Option(chunk.trailer))
+                case (accumulatedEntity, chunk) => (accumulatedEntity._1 ++ chunk.data, None)
               }
             }
+            case entity:HttpEntity => {
+              entity.withoutSizeLimit.dataBytes.runFold(blank)(_ ++ _).map (_ -> None)
+            }
+          }).map {
+            case (entityBytes, trailerHeaders) =>
+
+              val combinedHeaders = headers ++ trailerHeaders.getOrElse(Seq.empty)
```

Comment:
maybe you should try this (more scala way):
`trailerHeaders.fold(headers)(headers ++ _)`
this way you won't create another object in case there are no footers.

Reply:
Yeah! Much nicer! Done

Reply:
If you want it more cryptic (and more scala style) you can do:
`trailerHeaders.fold(headers)(_.++)`